src/frame/worker/crypto/recrypt/RecryptWasm.ts

```diff
@@ -125,13 +125,39 @@ const rotateUsersPrivateKey = (userPrivateKey: Uint8Array): Future<Error, {newPr
 /**
  * Calls rotateUsersPrivateKey, in the case that rotateUsersPrivateKey generates an augmentationFactor of zero or subtractPrivateKeys
  * results in zero an error is returned. This error is handled by calling rotateUsersPrivateKey again in an attempt produce valid results.
- * @param userPrivateKey
  */
-
 export const rotateUsersPrivateKeyWithRetry = (userPrivateKey: Uint8Array): Future<Error, {newPrivateKey: Uint8Array; augmentationFactor: Uint8Array}> => {
     return rotateUsersPrivateKey(userPrivateKey).handleWith(() => rotateUsersPrivateKey(userPrivateKey));
 };
 
+/**
+ * Generates a new key pair and takes the private key as the new group private key, The existing private key is then subtracted from the newly
+ * generated private key as the augmentation factor.
+ */
+const rotateGroupPrivateKey = (ExistingGroupPrivateKey: Uint8Array): Future<Error, {newPrivateKey: Uint8Array; augmentationFactor: Uint8Array}> => {
+    return generateKeyPair().flatMap(({privateKey}) => {
+        if (isBufferAllZero(privateKey)) {
+            return Future.reject(new Error("Key rotation failed."));
+        }
+        const newPrivateKey = RecryptApi.hash256(privateKey);
```

Comment:
Why is this necessary here to rotate a groups private key but it isn't necessary when we do user key rotation?